File: object_relations/app/models/customer.rb
# Customer initializes with a name
# Customer has many reviews
# Customer has many restaurants through reviews

class Customer
  attr_accessor :name, :reviews
  @@all = []

  def initialize(name)
    @name = name
    @reviews = []
    @@all << self
  end

  def add_review(review)
    @reviews << review
    review.customer = self
  end

  def self.all=(all)
    @@all = all
  end

  def self.all
    @@all
  end

  def self.find_by_name(name)
    @@all.find do |customer|
      customer.name == name
    end
  end
end

# customer = customer.add_review('it was good', Restaurant.new)
# a new review tied ot the restaurant
# and tied to the customer

File: javascript/models/customer.js
class Customer {
  constructor(name) {
    this.name = name;
    this.reviews = [];
    Store.customers.push(this);
  }

  all() {
    return Store.customers;
  }

  assignReview(review) {
    let customerName = findCustomer(review.customer.name);
    if customerName === this.name {
      Store.customers.reviews.push(review)
    }
  }

  findCustomer(name) {
    return Store.customers.find(customer => { return customer.name === name; })
    // return Store.customers.filter(function(obj) {
    //   return obj.name === name;
    // })
  }
}
